nodejs/awsx/apigateway/api.ts

```diff
@@ -553,6 +553,35 @@ export class API extends pulumi.ComponentResource {
             body: swaggerString,
         }, { parent: this });
 
+        // Account for all potential REST API Args that should trigger a redeployment
+        const argsHash = pulumi
+          .all([
+            this.restAPI.apiKeySource,
+            this.restAPI.binaryMediaTypes,
+            this.restAPI.endpointConfiguration,
+            this.restAPI.minimumCompressionSize,
+            this.restAPI.policy,
+            swaggerString,
+          ])
+          .apply(
+            ([
+              apiKey,
+              binaryMediaTypes,
+              endpointConfig,
+              minimumCompression,
+              policy,
+              swagger,
+            ]) =>
+              JSON.stringify({
+                apiKey,
+                binaryMediaTypes,
+                endpointConfig,
+                minimumCompression,
+                policy,
+                swagger,
+              }),
+          );
```

Comment:
1. because of the huge size of this, could you consider extracting a helper function?
2. alternatively, we can do less agressive wrapping  i.e. the following would be fine:

```.apply(([apiKey, binaryMediaTypes, endpointConfig, minimumCompression, policy, swagger]) =>```

This would also match the style we tend to use elsewhere!

Reply:
Thanks for the feedback, I've updated the code in alignment with your second recommendation (in order to be consistent with the rest of the codebase).